components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/LoginFlowAIService.java

```diff
@@ -0,0 +1,240 @@
+        } catch (AIClientException e) {
+            throw handleClientException(e);
+        }
+    }
+
+    /**
+     * Get login flow AI generation result.
+     *
+     * @param operationId Operation ID of the login flow AI generation.
+     * @return LoginFlowResultResponse.
+     */
+    public LoginFlowResultResponse getLoginFlowAIGenerationResult(String operationId) {
+
+        try {
+            Object generationResult = ApplicationManagementServiceHolder.getLoginFlowAIManagementService()
+                    .getAuthenticationSequenceGenerationResult(operationId);
+            LoginFlowResultResponse response = new LoginFlowResultResponse();
+            Map<String, Object> generationResultMap = (Map<String, Object>) generationResult;
+            response.setStatus(getStatusFromResult(generationResultMap));
+            if (!((Map<?, ?>) generationResult).containsKey(AI_RESPONSE_DATA_KEY)) {
+                throw new AIServerException(ERROR_CODE_ERROR_GETTING_LOGINFLOW_AI_RESULT_STATUS.getMessage(),
+                        ERROR_CODE_ERROR_GETTING_LOGINFLOW_AI_RESULT_STATUS.getCode());
+            }
+            Map<String, Object> dataMap = (Map<String, Object>) generationResultMap.get(AI_RESPONSE_DATA_KEY);
+            response.setData(dataMap);
+            return response;
+        } catch (AIServerException e) {
+            throw handleServerException(e);
+        } catch (AIClientException e) {
+            throw handleClientException(e);
+        }
+    }
+
+    /**
+     * Get login flow AI generation status.
+     *
+     * @param operationId Operation ID of the login flow AI generation.
+     * @return LoginFlowStatusResponse.
+     */
+    public LoginFlowStatusResponse getLoginFlowAIStatus(String operationId) {
+
+        try {
+            Object generationStatus = ApplicationManagementServiceHolder.getLoginFlowAIManagementService()
+                    .getAuthenticationSequenceGenerationStatus(operationId);
+            LoginFlowStatusResponse response = new LoginFlowStatusResponse();
+            response.setOperationId(operationId);
+
+            response.status(convertObjectToMap(generationStatus));
+            return response;
+        } catch (AIServerException e) {
+            throw handleServerException(e);
+        } catch (AIClientException e) {
+            throw handleClientException(e);
+        }
+    }
+
+    private StatusEnum getStatusFromResult(Map<String, Object> resultMap)
+            throws AIServerException {
+
+        if (resultMap.containsKey(AI_RESPONSE_STATUS_KEY)) {
+            String status = (String) resultMap.get(AI_RESPONSE_STATUS_KEY);
+            if ("IN_PROGRESS".equals(status)) {
+                return StatusEnum.IN_PROGRESS;
+            } else if ("COMPLETED".equals(status)) {
+                return StatusEnum.COMPLETED;
+            } else if ("FAILED".equals(status)) {
+                return StatusEnum.FAILED;
+            }
```

Comment:
Use fromValue in the enum instead. Catch the illegal arg exception and throw out the relevant exception